pyproject.toml

```diff
@@ -0,0 +1,29 @@
+[build-system]
+requires = [
+    "setuptools>=42",
+    "wheel",
+]
+build-backend = "setuptools.build_meta"
```

Comment:
This does activate the PEP 517 build system - this shouldn't affect anything, is the future of Python packaging, and may become the default after pip 21.3. It also works correctly in an environment without wheel installed, while the old build-system does not. But wanted to point out that this is a related change.

Reply:
I'm all for it, but IMHO this should happen in a separate PR. We might also consider migrating to `setup.cfg`.

Reply:
Okay, that has to be a pre-req for this one, though. Simply adding this file changes the build process to PEP 517, and the default for this is `"build-backend = "setuptools.build_meta.__legacy__"` (and it doesn't force setuptools >= 42 - 40 and 41 have bad bugs in pyproject.toml support).

Reply:
Pulled it out into #456, allow addressed the setup.cfg point in #457.

Reply:
(Simply adding a blank pyproject.toml file causes PEP 517 to activate, as the build-system selection is dependent on the presence of files, so I can't use the nice mypy config without #456 ad a pre-req)

Reply:
TIL :) Somehow I was under the impression that was a bug fixed long ago, but it turns out to be intended behavior, https://github.com/pypa/pip/issues/9738#issuecomment-808052363 and https://github.com/pypa/pip/issues/8437#issuecomment-805313362.

I see you just submitted a PR for this, thanks a lot!

Reply:
I think I finally understand the reasoning based on the filesystems (due to long discussions about whether a setup.cfg-only package with no pyproject.toml is a valid package under PEP 517), though from a user standpoint I still think it would have been much better to have required pyproject.toml _and_ a `project.build-backend` entry to activate.

PS: It was a very fine detail, but technically it is not valid to have a setup.cfg without either a pyproject.toml or setup.py, since PEP 517 allows a builder to directly call setup.py instead of running the setuptools entrypoint (though most/all builders choose the latter), so a valid PEP 517 builder could be written that wouldn't handle a project.toml file without pyprojec.toml, so `build` and `pip` have static checks requiring one of the two files).